File: Lua/Servidor.lua
require('socket');

puerto = 80;

servidor = socket.bind('*', puerto);

io.write("[NW] > Esperando en la entrada: "..puerto.."\n");

con = servidor:accept();

io.write("[NW] > Conexion desde: ");

while True do
    mensaje = con:receive();
    io.write(mensaje .. "\n");
    io.write("[NW] > Tu respuesta: ");
    con:send(io.read() .. "\n");
end 

io.read();

File: Scala/Servidor.scala
import java.io.{BufferedReader, InputStreamReader, PrintWriter}
import java.net.{ServerSocket, Socket}

object Servidor extends App
{
  var serverSocket = ServerSocket = new ServerSocket(1027)

  while(true)
  {
    val socket: Socket = serverSocket.accept()
    val entrada: BufferedReader = new BufferedReader(new InputStreamReader(socket.getInputStream))
    val salida: PrintWriter = new PrintWriter(socket.getOutputStream)

    val consulta = String = null;
    val continuar = Boolean = true;

    while(continuar)
    {
      consulta = entrada.readLine()

      if (consulta != null)
      {
        println(consulta + " desde el cliente...")
        salida.println(consulta + " desde el servidor...")
        salida.flush()
        continuar = false
      }
    }

    entrada.close()
    salida.close()
    socket.close()

  }
  serverSocket.close()
}
